xarray/core/parallelcompat.py

```diff
@@ -0,0 +1,265 @@
+"""
+The code in this module is an experiment in going from N=1 to N=2 parallel computing frameworks in xarray.
+It could later be used as the basis for a public interface allowing any N frameworks to interoperate with xarray,
+but for now it is just a private experiment.
+"""
+import functools
+import sys
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+from importlib.metadata import EntryPoint, entry_points
+from typing import (
+    Any,
+    Callable,
+    Generic,
+    Optional,
+    TypeVar,
+    Union,
+)
+
+import numpy as np
+
+from xarray.core.pycompat import is_chunked_array
+
+T_ChunkedArray = TypeVar("T_ChunkedArray")
+
+# TODO importing TypeAlias is a pain on python 3.9 without typing_extensions in the CI
+# T_Chunks: TypeAlias = tuple[tuple[int, ...], ...]
+T_Chunks = Any
+
+
+# Only used for testing purposes, as a real entrypoint is hard to mock
+EXAMPLE_CHUNKMANAGERS: dict[str, type["ChunkManagerEntrypoint"]] = {}
+
+
+@functools.lru_cache(maxsize=1)
```

Comment:
```suggestion
@functools.lru_cache(maxsize=2)
```

for dask and cubed?

Reply:
I don't think so? My understanding here is that you only need `list_chunkmanagers` to run once, because it will run when xarray is imported, and load all entrypoints at that time.

(Also I copied this from the `list_engines()` function and that seems to work fine even though we have many `BackendEntrypoint` to list)

Reply:
`lru_cache` will store the return value of the function, which in this case is a dict containing any chunk manager found. `maxsize` only has an effect if you have parameters (no kwargs to make the example smaller):
```python
@lru_cache(maxsize=2)
def f(x):
    return x

f(1)  # not cached, cache becomes {(1,): 1}
f(1)  # cached, cache unchanged
f(2)  # not cached, cache becomes {(1,): 1, (2,): 2}
f(2)  # cached, cache unchanged
f(3)  # not cached, cache becomes  {(2,): 2, (3,): 3}
f(2)  # cached, cache unchanged
f(4)  # not cached, cache becomes  {(2,): 2, (4,): 4}
...
```
which means that the least recently used entry is dropped if the cache exceeds a certain size (hence, `lru_cache`).

In our case we don't have any parameters, so `maxsize` should not matter at all (if the cache is pre-allocated that would reduce the memory footprint, though).

Reply:
Okay makes sense. That means this line in the file storage backends is not doing anything then either.

https://github.com/pydata/xarray/blob/635b9d0f2db4cc3ff005983c4af46ef6801258e2/xarray/backends/plugins.py#L106